### PR TITLE
New sitemap-generation stuff

### PR DESCRIPTION
**Pull request by ljpearlman**

### CODE REVIEW

File: docs/cli/deriva-sitemap-cli.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# deriva-sitemap-cli
```
Comment (mikedarcy) [inline docs/cli/deriva-sitemap-cli.md]: You'll want to edit `docs/index.rst` and add a reference to this file in the "Command Line Interfaces" toctree.

File: docs/api/deriva.seo.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
+deriva.seo package
+=======================
```
Comment (mikedarcy) [inline docs/api/deriva.seo.rst]: I know it sounds like nit-picking but `spinx-autodoc` will actually emit a warning when the `=` line string is too long/short for the title. Lame, I know. Not a biggie for now.

File: deriva/seo/sitemap_builder.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+                        {
+                           "image_url": "https://myhost.org/hatrac/images/image2.jpg",
+                        }
+                    ]
+                 }
+              ]
+        """
+        spec["data"] = rows
+
+    def find_incoming_fkeys(self, target_table):
```
Comment (robes) [inline deriva/seo/sitemap_builder.py]: I'm a little unclear about the use for this? Is this different than the `Table.referenced_by` property in `ermrest_model`?

Comment (ljpearlman) [reply deriva/seo/sitemap_builder.py]: I didn't know that property existed! That's convenient.

**Pull request merged**